src/OSPSuite.Core/Domain/Model.cs

```diff
@@ -66,6 +69,22 @@ public IContainer Neighborhoods
       //total drug mass is a parameter defined under the compound molecule global property
       public virtual IParameter TotalDrugMassFor(string moleculeName) => Root?.EntityAt<IParameter>(moleculeName, Constants.Parameters.TOTAL_DRUG_MASS);
 
+      public string MoleculeNameFor(string quantityPath)
+      {
+         var quantity = Root?.EntityAt<IQuantity>(quantityPath.ToPathArray());
+         return MoleculeNameFor(quantity);
+      }
+
+      public string MoleculeNameFor(IQuantity quantity)
+      {
+         if (quantity == null)
+            return string.Empty;
+
+         return quantity.IsAnImplementationOf<IMoleculeAmount>() ?
+            quantity.Name :
+            quantity.ParentContainer?.Name;
```

Comment:
if `quantity `is not a molecule amount: shouldn't we check that `quantity.ParentContainer` is a molecule amount before returning `quantity.ParentContainer?.Name`? 
If the function was called for a quantity which is e.g. not molecule related (plottable parameter for example), then parent container would be not a molecule.

Reply:
This would be also interesting test for RE: add a `plottable parameter `as output and check what happens 😄